services/src/machine_learning/xgb_config_builder.rs

```diff
@@ -0,0 +1,1057 @@
+/// This module contains the configuration builder for xgboost.
+/// See [https://xgboost.readthedocs.io/en/stable/parameter.html](https://xgboost.readthedocs.io/en/stable/parameter.html) for more details on the parameters.
+/// The `BoosterParameters` struct is the endpoint, that is sent to xgboost.
+use derive_builder::Builder;
+use std::borrow::Borrow;
+use std::cmp::Ordering;
+use std::collections::HashMap;
+use std::default::Default;
+use std::fmt::{self, Display};
+
+use serde::{Deserialize, Serialize};
+use xgboost_rs::DMatrix;
+
+use crate::error::Result;
+
+// -----------------------------------------------------------------------------------------------
+// XGBoost Config Definitions
+// -----------------------------------------------------------------------------------------------
+
+#[derive(Builder, Clone, Debug, Default, Serialize, Deserialize, PartialEq)]
+#[builder(default)]
+pub struct BoosterParameters {
+    booster_type: BoosterType,
+
+    learning_params: LearningTaskParameters,
+
+    verbose: bool,
+
+    threads: Option<u32>,
+}
+
+impl BoosterParameters {
+    pub fn booster_type(&self) -> &BoosterType {
+        &self.booster_type
+    }
+
+    pub fn set_booster_type<T: Into<BoosterType>>(&mut self, booster_type: T) {
+        self.booster_type = booster_type.into();
+    }
+
+    pub fn learning_params(&self) -> &LearningTaskParameters {
+        &self.learning_params
+    }
+
+    pub fn set_learning_params<T: Into<LearningTaskParameters>>(&mut self, learning_params: T) {
+        self.learning_params = learning_params.into();
+    }
+
+    pub fn verbose(&self) -> bool {
+        self.verbose
+    }
+
+    pub fn set_verbose(&mut self, verbose: bool) {
+        self.verbose = verbose;
+    }
+
+    pub fn threads(&self) -> &Option<u32> {
+        &self.threads
+    }
+
+    pub fn set_threads<T: Into<Option<u32>>>(&mut self, threads: T) {
+        self.threads = threads.into();
+    }
+
+    pub fn as_string_pairs(&self) -> Result<Vec<(String, String)>> {
+        let mut v = Vec::new();
+
+        v.extend(self.booster_type.as_string_pairs());
+
+        // here could be an option unset, so it needs to be a result value
+        let learning_params_res = self.learning_params.as_string_pairs()?;
+        v.extend(learning_params_res);
+
+        v.push(("silent".to_owned(), (u8::from(!self.verbose)).to_string()));
```

Comment:
why don't you call it "silent" in the first place?

Reply:
297f87dc

---

Commit MSG:

refactor: the verbose parameter is now called silent, as it reflects the functionality better